pkg/controller/jobs/pod/pod_controller_test.go

```diff
@@ -1555,6 +1558,339 @@ func TestReconciler(t *testing.T) {
 					Obj(),
 			},
 			workloadCmpOpts: defaultWorkloadCmpOpts,
+			// On the second reconciliation, the generic reconciler will try to update reclaimable
+			// pods. Since that operation uses server-side apply, which the fake client doesn't
+			// support, we're skipping reconcile for the second pod.
+			skipReconcileForPods: map[string]struct{}{"pod2": {}},
+		},
+		"if there's not enough non-failed pods in the group, workload should not be recreated": {
```

Comment:
uhm... not obvious that this should be the behavior... but at the same time, it's very hard to delete a workload without also deleting the Pods, given the finalizer logic.

Maybe we can actually just remove this test.

Reply:
But that shouldn't happen either... there shouldn't be Failed Pods until the groups is admitted at least once in the past.

Reply:
Yes, the case here is: 

1. Pod group is created;
2. Reconciler creates and admits the workload;
3. Some of the pods in the group are failing;
4. WL is deleted therefore pod sets are gone;
5. The new workload shouldn't be recreated until the reconciler has all the necessary information about the pod group. The group is not admitted for the second time.

Like you said, it's an unlikely scenario. But I think it's worth a unit test.

Reply:
sg.